applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp

```diff
@@ -28,8 +28,19 @@ void AdjointFiniteDifferenceSpringDamperElement<TPrimalElement>::InitializeSolut
     // would be part of the element properties this transferring would be not necessary. The stiffness parameters
     // are needed by the primal element in order to compute later on the element stiffness matrix for the
     // adjoint problem and the sensitivity matrix as the element contribution to the pseudo-load.
-    this->pGetPrimalElement()->SetValue(NODAL_DISPLACEMENT_STIFFNESS, this->GetValue(NODAL_DISPLACEMENT_STIFFNESS));
-    this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, this->GetValue(NODAL_ROTATIONAL_STIFFNESS));
+    auto stiffness_available = false;
+    if (this->Has(NODAL_DISPLACEMENT_STIFFNESS)) {
+        const auto temp = this->GetValue(NODAL_DISPLACEMENT_STIFFNESS);
+        this->pGetPrimalElement()->SetValue(NODAL_DISPLACEMENT_STIFFNESS, temp);
+        stiffness_available = true;
+    }
+    if (this->Has(NODAL_ROTATIONAL_STIFFNESS)) {
+        const auto temp = this->GetValue(NODAL_ROTATIONAL_STIFFNESS);
+        this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, temp);
```

Comment:
I would create a `const&` of the adjoint element and then set it in the primal
like this:
~~~c++
const auto& r_const_this = *this;

this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, r_const_this.GetValue(NODAL_ROTATIONAL_STIFFNESS));

~~~

Reply:
Thank you for the suggestion. I will try it.